doc/modeling/indices.md

```diff
@@ -0,0 +1,42 @@
+# Indices
+
+## PostgreSQL covering indices (INCLUDE)
+
+Since version 11, PostgreSQL supports [covering indexes](https://paquier.xyz/postgresql-2/postgres-11-covering-indexes), which allow you to include "non-key" columns in your indices. This allows you to perform index-only scans and can provide a significant performance boost:
+
+```c#
+protected override void OnConfiguring(DbContextOptionsBuilder builder)
+{
+    modelBuilder.Entity<Blog>()
+        .ForNpgsqlHasIndex(b => b.Id)
+        .ForNpgsqlInclude(b => b.Name);
+}
+```
+
+This will create an index for searching on `Id`, but containing also the column `Name`, so that reading the latter will not involve accessing the table. The SQL generated is as follows:
+
```

Comment:
Can we add the code sample before merging?

Reply:
Oops, I left that one out - fixed.